sdk/identity/identity/src/credentials/azurePipelinesServiceConnectionCredential.ts

```diff
@@ -0,0 +1,168 @@
+    return this.clientAssertionCredential.getToken(scopes, options);
+  }
+
+  /**
+   *
+   * @param oidcRequestUrl - oidc request url
+   * @param systemAccessToken - system access token
+   * @returns OIDC token from Azure Pipelines
+   */
+  private async requestOidcToken(
+    oidcRequestUrl: string,
+    systemAccessToken: string
+  ): Promise<string> {
+    logger.info("Requesting OIDC token from Azure Pipelines...");
+    logger.info(oidcRequestUrl);
+
+    const httpClient = createDefaultHttpClient();
+
+    const request = createPipelineRequest({
+      url: oidcRequestUrl,
+      method: "POST",
+      headers: createHttpHeaders({
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${systemAccessToken}`,
+      }),
+    });
+
+    const response = await httpClient.sendRequest(request);
+    const text = response.bodyAsText;
+    if (!text) {
+      throw new AuthenticationError(
+        response.status,
+        `${credentialName}: Authenticated Failed. Received null token from OIDC request.`
+      );
+    }
+    const result = JSON.parse(text);
+    if (result?.oidcToken) {
+      return result.oidcToken;
+    } else {
+      throw new AuthenticationError(
+        response.status,
+        `${credentialName}: Authentication Failed. oidcToken field not detected in the response. Response = ${JSON.stringify(
+          result
+        )}`
+      );
+    }
+  }
+
+  /**
+   * Ensures all system env vars are there to form the request uri for OIDC token
+   * @returns void
+   * @throws CredentialUnavailableError
+   */
+  private ensurePipelinesSystemVars(): void {
+    if (
+      process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI &&
+      process.env.SYSTEM_TEAMPROJECTID &&
+      process.env.SYSTEM_PLANID &&
+      process.env.SYSTEM_JOBID &&
+      process.env.SYSTEM_ACCESSTOKEN
+    ) {
+      return;
+    }
+    const missingEnvVars = [];
+    let errorMessage = "";
+    if (!process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI) {
+      missingEnvVars.push("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI");
+    }
+    if (!process.env.SYSTEM_TEAMPROJECTID) missingEnvVars.push("SYSTEM_TEAMPROJECTID");
+    if (!process.env.SYSTEM_PLANID) missingEnvVars.push("SYSTEM_PLANID");
+    if (!process.env.SYSTEM_JOBID) missingEnvVars.push("SYSTEM_JOBID");
+    if (!process.env.SYSTEM_ACCESSTOKEN) {
+      errorMessage +=
+        "\nPlease ensure that the system access token is available in the SYSTEM_ACCESSTOKEN value; this is often most easily achieved by adding a block to the end of your pipeline yaml for the task with:\n env: \n- SYSTEM_ACCESSTOKEN: $(System.AccessToken)";
+      missingEnvVars.push("SYSTEM_ACCESSTOKEN");
+    }
+    if (missingEnvVars.length > 0) {
+      throw new CredentialUnavailableError(
+        `${credentialName}: is unavailable.Please ensure that you're running this task in an Azure Pipeline, so that following missing system variable(s) can be defined- ${missingEnvVars.join(
```

Comment:
Missing space. Also, no need to include "Please" in these messages.
```suggestion
        `${credentialName}: is unavailable. Ensure that you're running this task in an Azure Pipeline, so that following missing system variable(s) can be defined- ${missingEnvVars.join(
```